core/src/org/javarosa/core/model/instance/TreeElement.java

```diff
@@ -998,25 +953,27 @@ public void populate(TreeElement incoming, FormDef f) {
                             children = new Vector();
                         }
                         this.children.insertElementAt(newChild, i + k + 1);
-                        newChild.populate((TreeElement)newChildren.elementAt(k), f);
+                        newChild.populate((TreeElement)newChildren.elementAt(k));
                     }
                     i += newChildren.size();
                 } else {
-
                     if (newChildren.size() == 0) {
                         child.setRelevant(false);
                     } else {
-                        child.populate((TreeElement)newChildren.elementAt(0), f);
+                        child.populate((TreeElement)newChildren.elementAt(0));
                     }
                 }
             }
-            for (int i = 0; i < incoming.getAttributeCount(); i++) {
-                String name = incoming.getAttributeName(i);
-                String ns = incoming.getAttributeNamespace(i);
-                String value = incoming.getAttributeValue(i);
 
-                this.setAttribute(ns, name, value);
-            }
+        }
+
+        // copy incoming element's attributes over
+        for (int i = 0; i < incoming.getAttributeCount(); i++) {
+            String name = incoming.getAttributeName(i);
+            String ns = incoming.getAttributeNamespace(i);
+            String value = incoming.getAttributeValue(i);
+
+            this.setAttribute(ns, name, value);
```

Comment:
The fix: moving this loop out of the else branch of the conditional above, which checks whether a node is a leaf. Attribute copying should happen regardless, right?